packages/enhanced/src/lib/sharing/resolveMatchedConfigs.ts

```diff
@@ -73,8 +72,18 @@ export async function resolveMatchedConfigs<T>(
         // module request prefix
         prefixed.set(request, config);
       } else {
+        let req = request;
+        //@ts-ignore
+        if ('import' in config && config.import && request !== config.import) {
+          req = config.import as string;
+        }
+        //@ts-ignore
+        if ('issuerLayer' in config && config.issuerLayer) {
+          //@ts-ignore
+          req = config.issuerLayer + req;
+        }
         // module request
-        unresolved.set(request, config);
+        unresolved.set(req, config);
```

Comment:
here i can construct contextInfo.issuerLayer from parent module - but we do not know layer it would become because this takes place in factorize of the module iteself, so i can make keys of ( issuerLayer )+ request. like (loader-layer)react as key.

then anything with issuerLayer of that, react request will match the request - which i think is correct use case here